src/ThemeCatalog.js

```diff
@@ -47,7 +47,8 @@ class ThemeHero extends Component {
    * catalog theme.
    */
   appendThemeStyle() {
-    if(this.themeStylesheetFilename) {
+    if(this.themeStylesheetFilename
+        && document.getElementById(this.themeStylesheetFilename) === null) {
```

Comment:
Isn't the id always `themeStylesheet`? Won't this always be `null`?

Reply:
This is not always null, in the event the user is flipping back and forth between two different components the fetch may be started twice, potentially resulting in 2 themes being added to the page. This catches that case.